hadoop-tools/hadoop-distcp/src/main/java/org/apache/hadoop/tools/mapred/CopyCommitter.java

```diff
@@ -635,7 +635,10 @@ private void concatFileChunks(Configuration conf, Path sourceFile,
         ++i;
       }
     }
+    long firstChunkLastModifiedTs = dstfs.getFileStatus(firstChunkFile)
+        .getModificationTime();
     dstfs.concat(firstChunkFile, restChunkFiles);
+    dstfs.setTimes(firstChunkFile, firstChunkLastModifiedTs, -1);
```

Comment:
I think you should update mtime after rename(). The HDFS rename updates mtime too. Not sure about other file system implementations but it's safe to assume that being the case.

Reply:
Sure @jojochuang , I can update mtime after rename(). However, for the better understanding, when I looked into it, it seems that mtime is updated for parent dirs rather than file itself.
For instance, renameTo updates mtime [here](https://github.com/apache/hadoop/blob/trunk/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSDirRenameOp.java#L222) and it internally updates mtime of srcParent and dstParent [here](https://github.com/apache/hadoop/blob/trunk/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSDirRenameOp.java#L747-L750)
Does this seem correct (unless I am missing something)?
Thanks

Reply:
Ah. interesting. Thanks for pointing out. I wasn't aware of that.
But I guess that makes sense for HDFS, since after rename, we update the quota for src and dest parent directory, so we do need to update mtime for them.

however, we need to support other file systems and it's not clear how they implement it.

Reply:
Makes sense, let me update this after rename()